### PR TITLE
Use hadoop conf and hive conf from catalog options

### PR DESCRIPTION
**Pull request by zhaomin1423**
### _Why are the changes needed?_

There are hdfs-site.xml, hive-site, etc in spark job classpath, but we should use hadoop conf and hive conf from catalog options.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: extensions/spark/kyuubi-spark-connector-hive/src/main/scala/org/apache/kyuubi/spark/connector/hive/HiveTableCatalog.scala [hunk 1]
```diff
@@ -115,6 +116,19 @@ class HiveTableCatalog(sparkSession: SparkSession)
...
 
+  private lazy val globalTempViewManager: GlobalTempViewManager = {
+    val globalTempDB = conf.getConf(GLOBAL_TEMP_DATABASE)
+    if (externalCatalog.databaseExists(globalTempDB)) {
+      throw KyuubiHiveConnectorException(
+        s"""
+           |$globalTempDB is a system preserved database, please rename your existing database
+           |to resolve the name conflict, or set a different value for
+           |${GLOBAL_TEMP_DATABASE.key}, and launch your Spark application again.
+         """.stripMargin.split("\n").mkString(" "))
```
Comment (pan3793) [inline extensions/spark/kyuubi-spark-connector-hive/src/main/scala/org/apache/kyuubi/spark/connector/hive/HiveTableCatalog.scala:127]: use 
```
"xxx " +
"xxx " +
"xxx"
```

File: extensions/spark/kyuubi-spark-connector-hive/src/main/scala/org/apache/kyuubi/spark/connector/hive/write/HiveWriteHelper.scala [hunk 2]
```diff
@@ -47,7 +47,7 @@ object HiveWriteHelper extends Logging {
   private val hiveScratchDir = "hive.exec.scratchdir"
 
   def getExternalTmpPath(
-      sparkSession: SparkSession,
+      externalCatalog: ExternalCatalogWithListener,
```
Comment (pan3793) [inline extensions/spark/kyuubi-spark-connector-hive/src/main/scala/org/apache/kyuubi/spark/connector/hive/write/HiveWriteHelper.scala:50]:
```suggestion
      externalCatalog: ExternalCatalog,
```

**Review by pan3793 — APPROVED** [review]
LGTM, one minor thing

File: extensions/spark/kyuubi-spark-connector-hive/src/main/scala/org/apache/kyuubi/spark/connector/hive/HiveTableCatalog.scala [hunk 3]
```diff
@@ -120,11 +120,9 @@ class HiveTableCatalog(sparkSession: SparkSession)
...
     if (externalCatalog.databaseExists(globalTempDB)) {
       throw KyuubiHiveConnectorException(
-        s"""
-           |$globalTempDB is a system preserved database, please rename your existing database
-           |to resolve the name conflict, or set a different value for
-           |${GLOBAL_TEMP_DATABASE.key}, and launch your Spark application again.
-         """.stripMargin.split("\n").mkString(" "))
+        s"$globalTempDB is a system preserved database, please rename your existing database to " +
+          s"resolve the name conflict, or set a different value for ${GLOBAL_TEMP_DATABASE.key}, " +
+          s"and launch your Spark application again.")
```
Comment (pan3793) [inline extensions/spark/kyuubi-spark-connector-hive/src/main/scala/org/apache/kyuubi/spark/connector/hive/HiveTableCatalog.scala:125]:
```suggestion
          "and launch your Spark application again.")
```

File: extensions/spark/kyuubi-spark-connector-hive/src/main/scala/org/apache/kyuubi/spark/connector/hive/HiveTableCatalog.scala (hunk 1 above)
Comment (zhaomin1423) [reply extensions/spark/kyuubi-spark-connector-hive/src/main/scala/org/apache/kyuubi/spark/connector/hive/HiveTableCatalog.scala]: done, thanks.

**Comment by pan3793** [comment]
Thanks, merged to master/1.7

**Pull request closed without merging**